dev/com.ibm.ws.config.schemagen/src/com/ibm/ws/config/schemagen/internal/Generator.java

```diff
@@ -141,13 +142,18 @@ int createSchema(String[] args) {
                     generate(smtp.getMetatypeInformation());
                     break;
                 case HELP_ACTION:
-                    // Only show command-line-style brief usage -help or --help invoked from command line
-                    System.out.println(MessageFormat.format(options.getString("briefUsageScript"), SCRIPT_NAME));
-                    System.out.println();
+
+                    showPurpose();                    
+                    showBriefUsage();
                     showUsageInfo();
 
                     rc = ReturnCode.OK;
                     break;
+                case NO_ARGUMENT:
+
+                    showPurpose();                    
+                    showBriefUsage();
+                    
```

Comment:
If `rc = generatorOptions.processArgs(args);` returns `22` (NO_ARGUMENT), what should the actual return code from the generator be? The original test was expecting a zero return code ... this is now dropping through and returning `20` (BAD_ARGUMENT).